lib/plugins/aws/package/compile/events/lib/ensureApiGatewayCloudWatchRole.js

```diff
@@ -1,13 +1,30 @@
 'use strict';
 
 const { memoize } = require('lodash');
+const BbPromise = require('bluebird');
 const { addCustomResourceToService } = require('../../../../customResources');
 
 module.exports = memoize(provider => {
   const cfTemplate = provider.serverless.service.provider.compiledCloudFormationTemplate;
+  const logicalId = provider.naming.getApiGatewayAccountCloudWatchRoleResourceLogicalId();
   const customResourceLogicalId = provider.naming.getCustomResourceApiGatewayAccountCloudWatchRoleResourceLogicalId();
   const customResourceFunctionLogicalId = provider.naming.getCustomResourceApiGatewayAccountCloudWatchRoleHandlerFunctionLogicalId();
 
+  // There may be a specific role ARN provided in the configuration
+  const config = provider.serverless.service.provider;
+  const restApi = config.logs && config.logs.restApi;
+  const configuredRoleArn = restApi && restApi.role;
+
+  if (configuredRoleArn) {
+    cfTemplate.Resources[logicalId] = {
+      Type: 'AWS::ApiGateway::Account',
+      Properties: {
+        CloudWatchRoleArn: configuredRoleArn,
```

Comment:
I think the proper way to handle that, would be to pass this into a custom resource, so the ARN is used instead of default one (and in that case we do not confirm on role polices).

Problem is that it's a region-wide setting. I'm not exactly sure how it behave with setup as proposed, when e.g. many stacks are deployed.

1. stack A is deployed, `CloudWatchRoleArn` is setup
2. stack B is deployed `CloudWatchRoleArn` is setup
3. stack A is removed. Will `CloudWatchRoleArn` setting remain untouched, or will it be cleared? If it's the latter, it'll break things for stack B.

When running it through custom resource, we may ensure given role is assigned on stack creation, and do not act on stack update and removal (it's how currently custom resource behaves)